kubernetes/job-db-migrate-production.tmpl

```diff
@@ -0,0 +1,20 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: panoptes-migrate-db-production
+spec:
+  template:
+    spec:
+      containers:
+      - name: panoptes-migrate-db-production
+        image: zooniverse/panoptes:__IMAGE_TAG__
+        command: ["/rails_app/migrate.sh"]
+        envFrom:
+        - secretRef:
+            name: panoptes-common-env-vars
+        - secretRef:
+            name: panoptes-production-env-vars
+        - configMapRef:
+            name: panoptes-production-shared
+      restartPolicy: Never
+  backoffLimit: 4
```

Comment:
I was assuming this would be 1. Are we retrying 4 times in case of transient failures? if so should we stop after 2?

Reply:
It uses an [exponential backoff interval](https://kubernetes.io/docs/concepts/workloads/controllers/job/#pod-backoff-failure-policy), so I think it makes sense to try a few times in case of temporary problems that take a bit longer to resolve themselves. Is there a down side to trying multiple times? I guess it's not ideal to retry a large migration that puts a heavy load on the DB for a while before failing.

Reply:
> I guess it's not ideal to retry a large migration that puts a heavy load on the DB for a while before failing.

I can see this happening if we're doing data changes that violate the schema (uniq index etc) in the migration but tbh that's bad practice generally and something to be avoided - looking at me https://github.com/zooniverse/panoptes/blob/d6f4fbd32761217d93a127c60e7fb7dbc5899c1a/db/migrate/20200327164455_add_mark_remove_column_to_recents.rb#L10

I'd be more inclined to report the failure so we can investigate and re-run the job (manually or via different jenkins job). If we find we need the retry behaviour for transient failures we can up the `backoffLimit` here at a later date.